modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/translator/DeepLClient.java

```diff
@@ -0,0 +1,133 @@
+package com.liferay.translation.translator.deepl.internal.translator;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.servlet.HttpHeaders;
+import com.liferay.portal.kernel.url.URLBuilder;
+import com.liferay.portal.kernel.util.ContentTypes;
+import com.liferay.portal.kernel.util.Http;
+import com.liferay.translation.translator.deepl.internal.constants.DeepLConstants;
+import com.liferay.translation.translator.deepl.internal.model.SupportedLanguage;
+import com.liferay.translation.translator.deepl.internal.model.TranslateResponse;
+import com.liferay.translation.translator.deepl.internal.util.JSONUtil;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * @author Yasuyuki Takeo
+ */
+@Component(
+    immediate = true,
+    service = DeepLClient.class
+)
+public class DeepLClient {
+
+  /**
+   * Translate text
+   */
+  public TranslateResponse execute(String authKey, String text, String sourcelang, String targetLang, String url)
+      throws IOException {
+    String rawRes = _fetch(authKey, text, sourcelang, targetLang, url);
+
+    return JSONUtil.toObject(rawRes, TranslateResponse.class);
+  }
+
+  protected String _fetch(
+      String authKey, String text, String sourcelang, String targetLang, String url) throws IOException {
```

Comment:
Fully spell variable names and use proper capitalization; e.g. `sourcelang` -> `sourceLanguageId`.